hazelcast/src/main/java/com/hazelcast/internal/management/ConsoleCommandHandler.java

```diff
@@ -104,5 +101,15 @@ public void println(Object obj) {
         public void print(Object obj) {
             buffer.append(String.valueOf(obj));
         }
+
+        @Override
+        protected void handleExit() {
+            print("'exit' is not allowed!");
```

Comment:
I would prefer to not use inheritance to get difference in behaviour. Assume there is another orthogonal setting that changes the behaviour of one other thing in this handler. This would already need 4 classes and complicated instantiation logic to cover all possible combinations. 

Personally I would go for feature switches in form off an `enum`. When the handler for the console is created one would pass the right switches. This can smoothly be extended with more switches whether they are orthogonal or not.

Reply:
I think inheritance is more appropriate for this one, as it's really a specialized and more restricted version of the original that is intended to be used for commands coming in from MC.